### ISSUE TITLE
Impediments graph

### ISSUE THREAD

**Issue by g4s8** [zerocracy/farm#1110]
There are a lot of impediments which registered without any reason, just `waiting`.
I think we can toughen the rule: allow to register impediment only for task in scope, e.g. if performer is waiting for PR review he can say `waiting for #pr` or if waiting for documentation he can submit a ticket in project to get put it into WBS and then say `waiting for #issue`. 
It will help to fit [QA rules](http://www.zerocracy.com/policy.html#42):
> All impediments registered in the ticket have explanations of the reasons and links to related tickets, if necessary

Also it makes possible to build impediment graph to analyze it and show messages in project chat, like: "This taks is blocking too many other tasks, consider boost it"; also we can show this graph on impediments page in project.

@yegor256 what do you think about it?

**Comment by amihaiemil** [comment]
@g4s8 If I may intervene: 

I wouldn't add any more restrictions/punishments to 0crat since it already is quite restrictive.
It has a few loopholes, which are good, they "let the steam off"... some of them are so obvious that I wonder if Yegor allowed them on purpose. The point is, I wouldn't remove these loopholes because then it would become quite inbarrable, it will make people lose all motivation (I lost mine for a while, due to this reason :)) 

If you want to revise some rules, I would redefine the number of defects that a CR has to find in code: it is almost imposible to find 3 reasonable defects in a PR from an experienced contributor (I lost a payment because of this), and I don't think it's ok to just require variable names changes (it's only a waste of time).

**Comment by g4s8** [comment]
@amihaiemil it can be not so restrictive, the idea is to make a graph of impediments to analyze it. I looked at Zerocracy WBS and wondered how many tasks are waiting for nothing (performer puts `waiting` without reason), 
and I can't see which tasks are blocked by another task or vice versa.

What about CR rules, you are speaking about "good quality" rules: if you find 3 defects - you will get 5 minutes extra for `REV` task (20 minutes), if you find only one - you will get 15 minutes, if zero - no payment.

**Comment by g4s8** [comment]
@yegor256 @amihaiemil  maybe we can make election rank to put on top these jobs which are blocking other jobs, e.g.: if job `#1` is blocked by `#2` and `#2` is blocked by `#3` this rank will calculate "impediment weight" = 2 and put job `#3` on top.

**Comment by amihaiemil** [comment]
@g4s8 Sure, it makes sense, but won't we have the Githbu API quota problem again? The same as we have with the Github Milestone Rank?

**Comment by yegor256** [comment]
@g4s8 this is relevant: #1239

**Comment by yegor256** [comment]
@g4s8 I like the idea of the election rule, to put "blockers" on top of other tasks. I like the idea of making impediments more specific and requiring developers to say what exactly a task is blocked by (which ticket).

**Comment by g4s8** [comment]
So let's enforce the reason first and try to match the reason by pattern: `waiting for #(?\d+)`: if pattern matched (e.g. `waiting for #42`), then register impediment with type `order`:
```xml
<impediment type="order">gh:test/test#123</impediment>
```
else (e.g. `waiting for answer`), with `unknown` type, like now implemented.

**Comment by 0crat** [comment]
Job [#1110](https://github.com/zerocracy/farm/issues/1110) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)
